src/test/java/com/connexta/store/CreateDatasetTests.java

```diff
@@ -89,9 +102,40 @@ public void testBadAcceptVersion() throws Exception {
   }
 
   @Test
-  @Disabled("TODO")
-  public void testCantReadAttachment() {
-    // TODO verify 400
+  void testCantReadAttachment() throws IOException {
+    MultipartFile file = spy(newValidMultipartFile());
+    doThrow(new IOException("Cannot read attachment")).when(file).getInputStream();
+    assertBadRequest(file);
+    verify(file).getInputStream();
+  }
+
+  @Test
+  void testFileTooLarge() {
```

Comment:
❓ What happens when a negative file size is passed in?

Reply:
Size comes from the stream and streams cannot have a negative length.